src/sentry/incidents/logic.py

```diff
@@ -994,7 +1009,17 @@ def update_alert_rule(
                 "time_window", timedelta(seconds=snuba_query.time_window)
             )
             updated_query_fields.setdefault("event_types", None)
-            updated_query_fields.setdefault("resolution", timedelta(seconds=snuba_query.resolution))
+            if detection_type and detection_type == AlertRuleDetectionType.DYNAMIC:
```

Comment:
this is the piece that was missing from the [previous PR](https://github.com/getsentry/sentry/pull/77364) that set the resolution to the time window, it would be overwritten here